h2o-core/src/main/java/water/Key.java

```diff
@@ -299,7 +299,37 @@ public static String rand() {
   public static <P extends Keyed> Key<P> makeUserHidden(String s) {
     return make(s,DEFAULT_DESIRED_REPLICA_FACTOR,HIDDEN_USER_KEY, false);
   }
-
+  
+  /** Factory making an random Key from two Strings
+   * 
+   * @param prefix - prefix of the new key. Should not be null. If it contains non-word characters, 
+   *               they are converted to underscores.
+   * @param postfix - postfix of the new key. Should not be null. If it contains non-word characters,
+   *                they are converted to underscores.
+   *
+   *  @return Desired Key that looks like prefix_RANDOM_postfix 
+   */
+  public static <P extends Keyed> Key<P> makeRandom(String prefix, String postfix) {
```

Comment:
(Optional) make args final for absolute perfection.

Reply:
Let's make it even better :-D